core/domain/suggestion_registry.py

```diff
@@ -421,24 +424,26 @@ def __init__(
             exp_domain.ExplorationChange(change)
         )
         self.score_category = score_category
-        # In BaseSuggestion, language_code is defined with only string type
-        # but here language_code is of Optional[str] type because language_code
-        # can accept None values as well. So, due to this conflict in types
-        # MyPy throws an `Incompatible types in assignment` error. Thus to
-        # avoid the error, we used ignore here.
+        # Here we use MyPy ignore because in BaseSuggestion, language_code
+        # is defined with only string type but here language_code is of
```

Comment:
Shouldn't BaseSuggestion's language_code be fixed, then?

Reply:
Then it will be optional for every sub-class. And not all sub-classes accept language_code as None.

Reply:
I'm not introducing Opitional[str] in the base class's language_code because:
1. Only `SuggestionEditStateContent` can accept None value for language_code https://github.com/oppia/oppia/blob/20af54e0574a4f99690f15f2fbb97f2a397f22f9/core/domain/suggestion_registry.py#L479
and all other classes cannot accept None values for language_code https://github.com/oppia/oppia/blob/20af54e0574a4f99690f15f2fbb97f2a397f22f9/core/domain/suggestion_registry.py#L698
https://github.com/oppia/oppia/blob/20af54e0574a4f99690f15f2fbb97f2a397f22f9/core/domain/suggestion_registry.py#L951

2. If we use Optional[str] in the base class, then in all sub-classes language_code is going to be Optional[str] and we have to use `if else raise exception` unnecessary ( even though we know that in particular sub-class language_code is not going to be None) -- you can see an example that I have created for this case https://mypy-play.net/?mypy=latest&python=3.10&flags=strict%2Cdisallow-any-expr&gist=ac114a174340b007d549a4eca10e3e99

Reply:
ok